ci/customization/customize_doc.py

```diff
@@ -20,38 +20,23 @@
 FA_TAG_ID = "rapids-fa-tag"
 
 
-class r_versions(str):
-    def compare(self, other: str) -> int:
-        yearA, monthA = map(int, self.split("."))
-        yearB, monthB = map(int, other.split("."))
-
-        if yearA < yearB or (yearA == yearB and monthA < monthB):
-            return -1
-        elif yearA == yearB and monthA == monthB:
-            return 0
-        else:
-            return 1
-
-    def is_less_than(self, other: str) -> bool:
-        return self.compare(other) == -1
-
-    def is_greater_than(self, other: str) -> bool:
-        return self.compare(other) == 1
-
-
 def get_version_from_fp(*, filepath: str, versions_dict: dict):
     """
     Determines if the current HTML document is for legacy, stable, or nightly versions
     based on the file path
     """
     match = re.search(r"/(\d?\d\.\d\d)/", filepath)
-    version_number_str = r_versions(match.group(1))
-    version_name = "stable"
-    if version_number_str.is_greater_than(versions_dict["stable"]):
-        version_name = "nightly"
-    if version_number_str.is_less_than(versions_dict["stable"]):
-        version_name = "legacy"
-    return {"name": version_name, "number": version_number_str}
+    version_number_from_filepath = match.group(1)
+
+    # given a version number like "25.10", figure out the corresponding version name like "stable", "nightly", or "legacy"
+    for version_name, version_number in versions_dict.items():
+        if version_number == version_number_from_filepath:
+            return {"name": version_name, "number": version_number_from_filepath}
```

Comment:
I started the changes in this function to account for projects which don't have requested versions of all of `nightly`, `stable`, and `legacy. For example, for `rapidsmpf` we only host `nightly` API docs today:

https://github.com/rapidsai/docs/blob/5c2a36d408da9960d880602152460b1d11c34591/_data/docs.yml#L170-L179

Because the version number we're comparing here is just `{major}.{minor}` (doesn't need to account for hotfix versions like `25.10.01`), I think we can rely on always having **exact** matches.

And so don't need the `r_versions` class and its version-comparison logic.